### PR TITLE
Makefile: add direnv-check to verify it is loaded

### PR DESCRIPTION
**Pull request by jakubgs**
Result:

image

Also added a way for users to provide their own method of loading the Vault token.

### CODE REVIEW

**Review by siddarthkay — APPROVED** [review]
I tested this locally and it works well

File: .envrc.secrets [hunk 1]
```diff
@@ -3,3 +3,7 @@ export VAULT_CLIENT_CERT=./ansible/files/vault-client-user.crt
 export VAULT_CLIENT_KEY=./ansible/files/vault-client-user.key
 export VAULT_ADDR=https://vault-api.infra.status.im:8200
 export CONSUL_HTTP_TOKEN=$(pass services/consul/tokens/terraform)
+# Provide a script in your PATH matching this name to load the token.
+if command -v vault_token_provider >/dev/null; then
+    export VAULT_HTTP_TOKEN=$(vault_token_provider)
```
Comment (apentori) [inline .envrc.secrets:8]: The var should be `VAULT_TOKEN`

**Review by apentori — CHANGES_REQUESTED** [review]

**Review by yakimant — APPROVED** [review]
Beautiful!

**Review by markoburcul — APPROVED** [review]
Works like a charm.

**Review by apentori — APPROVED** [review]

**Pull request merged**